src/validators/enum_.rs

```diff
@@ -116,7 +117,7 @@ impl<T: EnumValidateValue> Validator for EnumValidator<T> {
                 },
                 input,
             ));
-        } else if let Some(v) = T::validate_value(py, input, &self.lookup, strict)? {
+        } else if let Some(v) = T::validate_value(py, input, &self.lookup, &self.class, strict)? {
             state.floor_exactness(Exactness::Lax);
             return Ok(v);
         } else if let Some(ref missing) = self.missing {
```

Comment:
So now that I look at this new point in the diff, I see that (and I regret forgetting this) that we already call `_missing_` here. But as we observe in the issue and this PR, simply calling `_missing_` is not enough because enum `__new__` has more complex logic which isn't encapsulated purely by `_missing_`.

I wonder if there is a case to have a new branch (before or after this one? not sure 🤔) which calls the enum type, with the logic going here instead of in `PlainEnumValidator`. Putting the logic here would also solve the special-cased enums like `IntEnum`, I think.

That does beg the question, though: if we add the case of calling the enum type here, do we need logic for `_missing_` at all? My intuition is that we don't, and we should try to phase out the `_missing_` logic.